gslib/tracker_file.py

```diff
@@ -554,14 +554,28 @@ def WriteDownloadComponentTrackerFile(tracker_file_name, src_obj_metadata,
 def _WriteTrackerFile(tracker_file_name, data):
   """Creates a tracker file, storing the input data."""
   try:
-    with os.fdopen(os.open(tracker_file_name, os.O_WRONLY | os.O_CREAT, 0o600),
-                   'w') as tf:
+    fd = os.open(tracker_file_name, os.O_WRONLY | os.O_CREAT | os.O_TRUNC,
+                 0o600)
```

Comment:
so the major change here is that you're using `os.O_TRUNC` to empty the file, yes?

Reply:
The major change for the entire PR is making `_UploadTrackerCallback` call this function instead of trying to write the tracker file directly.

For this particular change specifically, you are correct, the os.)_TRUNC is the major change. This was added because I saw some unpredictable behavior if the file is not emptied.